### PR TITLE
TST: adding tests of current clear behavior on ticks

### LINKED ISSUE #23839: [MNT]: Add tests to codify `ax.clear`
### Summary

As discussed on today's call.

We need to codify what `ax.clear` and `plt.cla` are meant to actually clear on an axes, versus setting on the axes that should remain static.  

For instance, it is pretty straight forward that clearing an axes should remove any data artists from the plot.  However, what other properties should be reset.  Labels?  Tick styling, grids?  Units on axes?   Locators and formatters?  The scale of the axises?  

If anyone with edit access wants, they should start a list here, or if folks don't have edit access they can add list items on the comments.  

Crossref: https://github.com/matplotlib/matplotlib/pull/23808

More reference from @timhoffm (https://github.com/matplotlib/matplotlib/pull/23808#pullrequestreview-1099932104):
- #20149 is the original issue, that started everything
- #20161 was the bugfix for above (v3.4.2)
- resulting in regressions #20721 and #20219.
- these were hot-fixed in #20826 (v3.4.3)
- #22587 (v3.6.0) claims to be a follow-up cleanup which claims to not change the outcome, but apparently we did not have enough tests.

### Proposed fix

_No response_

### PR DESCRIPTION
**Pull request by tacaswell**
closes #23839

This implements the suggestion from @timhoffm in https://github.com/matplotlib/matplotlib/issues/23806#issuecomment-1239069010

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -8898,3 +8898,41 @@ def test_axhvlinespan_interpolation():
...
+        "labeltop": True,
+        "labelleft": True,
+        "labelright": True,
+    }
+
+    for ax in [ax_test, ax_ref]:
+        ax.tick_params(axis=which, **target)
+
+    ax_test.clear()
+    ax_test.tick_params(axis=which, **target)
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py:8938]: I do not understand what this test is meant to catch. Unrolling the for loop, you do on `ax_test`
```
ax_test.tick_params(axis=which, **target)
ax_test.clear()
ax_test.tick_params(axis=which, **target)
```
which, irrespective of the `clear()` behavior, should be the same as only
```
ax_test.tick_params(axis=which, **target)
```
This is compared to 
```
ax_ref.tick_params(axis=which, **target)
```
Unless I'm mistaken somewhere above, this test is trivally true and doesn't actually test the clear behavior.

In https://github.com/matplotlib/matplotlib/issues/23806#issuecomment-1239069010 I had suggested a different pattern to catch the parameters that should survive `clear()`.

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -8898,3 +8898,43 @@ def test_axhvlinespan_interpolation():
     ax.axhline(1, c="C0", alpha=.5)
     ax.axhspan(.8, .9, fc="C1", alpha=.5)
     ax.axhspan(.6, .7, .8, .9, fc="C2", alpha=.5)
+
+
+@check_figures_equal(extensions=["png"])
+@pytest.mark.parametrize("which", ("x", "y"))
+def test_axes_clear_behavior(fig_ref, fig_test, which):
+    ax_test = fig_test.subplots()
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py:8906]:
```suggestion
        """Test that the given tick params are not reset by ax.clear()."""
    ax_test = fig_test.subplots()
```

File: lib/matplotlib/tests/test_axes.py [hunk 3]
```diff
@@ -8898,3 +8898,43 @@ def test_axhvlinespan_interpolation():
...
+    ax_ref = fig_ref.subplots()
+    target = {
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py:8908]:
```suggestion
    # the following tick params values are chosen to each create a visual difference
    # from their defaults 
    target = {
```

**Review by timhoffm — COMMENTED** [review]
Adding some comments makes it more clear what the test does and how it works.

**Comment by timhoffm** [comment]
Ping @tacaswell This is basically good to go. Only minor comments to make the test more understandable.

File: lib/matplotlib/tests/test_axes.py [hunk 4]
```diff
@@ -8903,8 +8903,11 @@ def test_axhvlinespan_interpolation():
 @check_figures_equal(extensions=["png"])
 @pytest.mark.parametrize("which", ("x", "y"))
 def test_axes_clear_behavior(fig_ref, fig_test, which):
+    """Test that the given tick params are not reset by ax.clear()."""
     ax_test = fig_test.subplots()
     ax_ref = fig_ref.subplots()
+    # the following tick params values are chosen to each create a visual difference
+    # from their defaults
```
Comment (ksunden) [inline lib/matplotlib/tests/test_axes.py:8910]:
```suggestion
    # from their defaults
```

Found the whitespace at the end of this line

**Review by timhoffm — APPROVED** [review]

**Pull request merged**